### ISSUE TITLE
Why showing tooltips on button focus?

### ISSUE THREAD

**Issue by Reinmar** [ckeditor/ckeditor5#12066]
We started showing tooltips for focused buttons in https://github.com/ckeditor/ckeditor5/issues/5581.

However, I don't understand why.

* Are there a11y reasons to do so? If so – what other apps do that because I briefly checked some and none did so.
* Such tooltips can negatively affect the UX. For instance, if you open the font color dropdown or alignment dropdown that's what you see:
<img width="616" alt="image" src="https://user-images.githubusercontent.com/156149/179184557-1f08e6ba-d527-429f-b3f3-996ceba03577.png"> 
<img width="420" alt="image" src="https://user-images.githubusercontent.com/156149/179184637-78b05319-1189-4a53-8b5b-43e0030c15d4.png">

What should we do about this?

---

If you'd like to see this improvement implemented, add a 👍 reaction to this post.

**Comment by Reinmar** [comment]
I had one example under my fingertips :D 

<img width="710" alt="image" src="https://user-images.githubusercontent.com/156149/179185607-5966a237-553d-4217-bdef-a4af1e047e44.png">

**Comment by Comandeer** [comment]
> If so – what other apps do that because I briefly checked some and none did so.

It will probably be soon a default behavior in Chromium → [https://bugs.chromium.org/p/chromium/issues/detail?id=829352](https://bugs.chromium.org/p/chromium/issues/detail?id=829352)

> Are there a11y reasons to do so?

Yes, because until recently only mouse users have access to the labels of the buttons. With this change, also keyboard users (and probably touch ones) have it. It's connected with [SC 3.3.2 Labels or instructions](https://www.w3.org/TR/WCAG21/#labels-or-instructions).

TBH I don't see the negative impact you refer to :thinking: The tooltip moves with the focus, so even if some button is slightly obscured by it, moving the focus would also move the tooltip, fixing the issue. The only issue I see in the screenshots you provided is the duplication of the label in case of the "Remove color" button – there is no need for a tooltip if the element has text content.

**Comment by Reinmar** [comment]
Decision:

*   Let's revert the tooltips change for now.
*   We can always bring it back with some small bandaids:
    *   Turning the alignment menu to horizontal
    *   Removing the tooltips from buttons that have a text
*   But let's take the time now before August to try to come up with a better solution, e.g. Slack-like or one that removes tooltips on mousemove wherever on the page (unless the button is hovered).

**Comment by Reinmar** [comment]
Followups:

*   #12068 
*   #12067

**Comment by Comandeer** [comment]
There is a [`:focus-visible` pseudoclass](https://developer.mozilla.org/en-US/docs/Web/CSS/:focus-visible) which can probably help us to some extent.
